### DISCUSSION TITLE
Todo Sample App

### DISCUSSION THREAD

**Discussion by Chuckh008** [Azure/azure-mobile-apps#449] [Q&A] [answered]
I have the  todo sample running on Android, using the Maui App. The Chrome browser is launched outside of the application to authenticate with Active Directory. Should the app launch an internal browser?

parentWindow = Platform.CurrentActivity;
IdentityClient = PlatformService.GetIdentityClient(parentWindow);

**Comment by adrianhall** [comment, marked as answer]
There is no internal browser on Android - it's all Chrome.  We use "custom tabs" functionality, which starts Chrome with a responder.  So what you are seeing is completely normal.

**Reply by Chuckh008** [reply]
So I understand correctly...The Todo app is suppose to launch the Chrome browser app outside of the running application and create a new tab for each new sign in.

Sorry, when I stated internal browser, I meant launch Chrome internally.

<img width="298" alt="Todo-Chrome-1" src="https://user-images.githubusercontent.com/45703080/185198305-e2e14271-5cf8-49cf-94ff-da82891b8f10.png">


<img width="296" alt="Todo-Chrome-Tabs" src="https://user-images.githubusercontent.com/45703080/185198375-3118587b-ee92-48b5-80b5-dcf4f24ea011.png">

Thanks Adrian
.

**Reply by adrianhall** [reply]
The code for that is driven by MSAL - the library doing the authentication.  The TodoApp delegates authentication to the MSAL library, so we don't get to decide how they have implemented it.
